src/ipvs/ip_vs_redirect.c

```diff
@@ -85,14 +85,13 @@ dp_vs_redirect_alloc(enum dpvs_fwd_mode fwdmode)
     return r;
 }
 
-void dp_vs_redirect_free(struct dp_vs_conn *conn)
+void dp_vs_redirect_free(struct dp_vs_redirect *redirect)
 {
-    if (conn->redirect) {
+    if (redirect) {
```

Comment:
Adding the check against "conn" is enough and no need to expose "conn->redirect = NULL" to outside.